src/librustdoc/passes/check_code_block_syntax.rs

```diff
@@ -47,51 +49,78 @@ impl<'a, 'tcx> SyntaxChecker<'a, 'tcx> {
         .unwrap_or(false);
         let buffer = buffer.borrow();
 
-        if buffer.has_errors || is_empty {
-            let mut diag = if let Some(sp) =
-                super::source_span_for_markdown_range(self.cx, &dox, &code_block.range, &item.attrs)
-            {
-                let warning_message = if buffer.has_errors {
+        if !(buffer.has_errors || is_empty) {
+            // No errors in a non-empty program.
+            return;
+        }
+
+        let local_id = match item.def_id.as_local() {
+            Some(id) => id,
+            // We don't need to check the syntax for other crates so returning
+            // without doing anything should not be a problem.
+            None => return,
+        };
+
+        let hir_id = self.cx.tcx.hir().local_def_id_to_hir_id(local_id);
+        let mark_with_text = code_block.syntax.is_none() && code_block.is_fenced;
+
+        // The span and whether it is precise or not.
+        let (sp, precise_span) = match super::source_span_for_markdown_range(
+            self.cx,
+            &dox,
+            &code_block.range,
+            &item.attrs,
+        ) {
+            Some(sp) => (sp, true),
+            None => {
+                (super::span_of_attrs(&item.attrs).unwrap_or_else(|| item.source.span()), false)
```

Comment:
FYI `span` is trivial, I think `or_else` is a little more verbose than it needs to be. https://doc.rust-lang.org/nightly/nightly-rustc/src/rustdoc/clean/types.rs.html#1629-1631
```suggestion
                (super::span_of_attrs(&item.attrs).unwrap_or(item.source.span()), false)
```